specs/agents/logging.md

```diff
@@ -31,11 +30,112 @@ agents can treat that as a synonym for `error` or `fatal`.
+
+* **Agent**: This block is mandatory and contains basic version and build date information.
+* **Environment**: This block is optional but for supportability reasons it should be provided.
+* **Configuration**: This block mandatory and contains a minimum set of relevant configuration values.
+
+### Agent
+
+On startup, all APM agents MUST log basic information regarding their technology (language, runtime),
+version information, and build date in a format chosen by the respective agent.
+
+This SHOULD be the very first log message that is created by an agent.
+
+Example:
+
+```text
+Elastic APM .NET Agent, version: 1.19.1-preview, build date: 2022-10-27 10:55:42 UTC
+```
+
+Agents SHOULD also report when they are running in a **non-production** scenario (e.g. a debug
+or pre-release build) using the `warning` logging level.
+
+Example:
+
+```text
+This a pre-release version and not intended for use in production environments!
+```
+
+### Environment
+
+Additionally, agents SHOULD report information about their environment (e.g. host, process, runtime).
+
+| Item | Description | Example |
+| - | - | - |
+| Process ID | The Process ID in decimal format. | `83606` |
+| Process Name | The executable image name or the full path to it.  | `w3wp.exe`, `/usr/local/share/dotnet/dotnet` |
+| Command Line | The full command line used to launch this process as available to the runtime. | `/Users/acme/some_app/bin/Debug/net7.0/some_app.dll foo=bar` |
+| Operating System | OS name and version in a human-readable format. | `macOS Version 12.6.1 (build 21G217)` |
+| CPU architecture | One of the well-known values from the table below. | `arm64` |
+| Host | The (optionally fully-qualified) host name. | `MacBook-Pro.localdomain` |
+| Time zone | The local time zone in UTC-offset notation. | `UTC+0200` |
+| Runtime | Name and version of the executing runtime. | `.NET Framework 4.8.4250.0`|
+
+**CPU Architecture:**
+
+| Value | Description |
+| - | - |
+| `amd64` | AMD64 |
+| `arm32` |ARM32 |
+| `arm64` |ARM64 |
+| `ia64` | Itanium |
+| `ppc32` | 32-bit PowerPC |
+| `ppc64` | 64-bit PowerPC |
+| `s390x` | IBM z/Architecture |
+| `x86` | 32-bit x86 |
+
+### Configuration
+
+The start of the configuration block MUST be denoted as such (e.g. `Agent Configuration:`).
+
+If a configuration file was used in the configuration process, its fully-qualified path
+SHOULD be logged.
+
+Configuration item names MUST be provided in normalized (lower-case, snake_case) notation.
+Configuration value strings MUST be printed in quotes (so accidental leading or trailing whitespace can be spotted).
+
+Agents MUST provide information about following essential configuration items:
+
+| Item | Needs masking | Example |
+| - | - | - | - |
+| `server_url` | no | `http://localhost:8200` |
+| `secret_token` | yes | `*****` |
+| `service_name` | no | `foo` |
+| `log_level` | no | `warning` |
+
+Additional configurations items MAY be logged after that as well.
```

Comment:
Could we simplify to logging all options that don't have default values?

Reply:
Did you mean:
a) Log the above MUST config options. Additionally, all non-default options MAY be logged.
b) All non-default options MUST be logged.

Reply:
b

Reply:
I wasn't sure if every agent currently has the infrastructure to:
* enumerate all config options
* generically mask those that need masking
(I know for a fact that e.g. the .NET agent does not provide this atm).

That's why I went for those few selected config options, made them a MUST requirement, and the rest a nice to have.

However, if we are OK with requiring the above-mentioned functionality from all agents, I agree with our suggestion.